main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -237,6 +242,24 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       }
     }
 
+    def Effect: Rule1[ParsedAst.Declaration] = {
+      def Head = rule {
+        Documentation ~ Modifiers ~ SP ~ keyword("eff") ~ WS ~ Names.Effect ~ optWS ~ TypeParams
+      }
+
+      def EmptyBody = namedRule("EffectBody") {
```

Comment:
I thought we were using these namedRule sparingly?

Reply:
Parallels Instance and Class.